api/handlers.go

```diff
@@ -70,6 +72,237 @@ func CollectionHandler(c *gin.Context) {
 	concludeRequest(c, resp, err)
 }
 
+func LessonOverviewHandler(c *gin.Context) {
+	var request LessonOverviewRequest
+	if c.Bind(&request) != nil {
+		return
+	}
+	db := c.MustGet("MDB_DB").(*sql.DB)
+	resp, err := getLessonOverviewsPage(db, request)
+	if err != nil {
+		NewInternalError(err).Abort(c)
+		return
+	}
+
+	var collectionIds []int64
+	var cuIds []int64
+	var collectionUids []string
+	itemsMap := make(map[string]*LessonsOverviewResponseItem)
+	for _, item := range resp.Items {
+		itemsMap[item.CollectionId] = item
+		collectionIds = append(collectionIds, item.internalCollectionId)
+		cuIds = append(cuIds, item.internalUnitId)
+		collectionUids = append(collectionUids, item.CollectionId)
+	}
+
+	if err = setI18ColNameDesc(db, request.Language, collectionIds, cuIds, resp.Items); err != nil {
+		NewInternalError(err).Abort(c)
+		return
+	}
+
+	if viewsResp, err := getViewsByCollectionIds(collectionUids); err != nil {
+		NewInternalError(err).Abort(c)
```

Comment:
In case we have error with feed API, do not trigger internal error. Just write log and continue.

Reply:
Done